examples/agents/granite/README.md

```diff
@@ -67,11 +67,49 @@ The [ibm_granite_bee](/examples/agents/ibm_granite_bee.ts) example agent is set
-    yarn run start examples/agents/ibm_granite_bee.ts <<< 'What is the spanish word for dog?'
-    yarn run start examples/agents/ibm_granite_bee.ts <<< 'What does the alias command do in the BASH shell?'
-    yarn run start examples/agents/ibm_granite_bee.ts <<< "What’s the largest technology company by market capitalization right now?"
-    yarn run start examples/agents/ibm_granite_bee.ts <<< "What’s the weather going to be like tomorrow in Sydney?"
+    yarn run start examples/agents/granite/granite_bee.ts <<< 'Who is the president of the USA?'
+    yarn run start examples/agents/granite/granite_bee.ts <<< 'What is the spanish word for dog?'
+    yarn run start examples/agents/granite/granite_bee.ts <<< 'What does the alias command do in the BASH shell?'
+    yarn run start examples/agents/granite/granite_bee.ts <<< "What’s the largest technology company by market capitalization right now?"
+    yarn run start examples/agents/granite/granite_bee.ts <<< "What’s the weather going to be like tomorrow in Sydney?"
    ```
 
    The example is configured with 2 available [tools](/docs/tools.md), these are OpenMeteo for facilitating weather related prompts and DuckDuckGo for facilitating search related prompts. The above example prompts demonstrate how each of these tools can be exercised.
+
+## 🤖 Granite wiki bee
+
+The [granite_wiki_bee](/examples/agents/granite/granite_wiki_bee.ts) example agent is set up to demonstrate more advanced usage of the bee agent framework, specifically
+performing information retrieval on tool output via the frameworks tool customization options. This agent is configured to use the `WikipediaTool` which enables the agent to
+search wikipedia and access articles. However the tool interface is extended so that the agent can specify both a wikipedia topic and a search query within the topic. The wikipedia tool output is chunked and piped into a `SimilarityTool` that calculates the cosine similarity between the query and the data chunks. Only the most relevant chunks are then provided to the agents memory.
+
+This example uses ollama exclusively.
+
+To get started you will need to pull `granite3-dense:8b` and `nomic-embed-text` (to perform text embedding).
+
+```shell
+ollama pull granite3-dense:8b
+ollama pull nomic-embed-text
+```
+
+Run the [granite_wiki_bee](/examples/agents/granite/granite_wiki_bee.ts) example:
+
+```shell
+yarn run start examples/agents/granite/granite_wiki_bee.ts
+```
+
+You will see the agent running and producing an answer similar to the following:
+
+> Agent 🤖 : The authors of the paper 'Attention is all you need' are Ashish Vaswani, Noam Shazeer, Niki Parmar, Jakob Uszkoreit, Llion Jones, Aidan Gomez, Lukasz Kaiser, and Illia Polosukhin. The paper has been cited more than 100,000 times as of 2024.
+
+The default prompt is the question `Who were the authors of the paper 'Attention is all you need' and how many citations does it have?`
```

Comment:
I use this example because its a seminal paper that has its own wikipedia article. I dont see an article specifically on the Granite Code Models paper.

Reply:
Ah yes, good point.  I was getting mixed up with the arxiv tool so the one you've chosen is likely to be much better.